File: com.example.text/java/MainActivity.java
package com.example.text;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        final TextView textComponent = findViewById(R.id.textlabel);
        Button button = findViewById(R.id.button);

        button.setOnClickListener(new View.OnClickListener() {
        
            @Override public void onClick(View v) {
                textComponent.setText("HI");
            }
        });
    }
}

File: com.example.text/kotlin/MainActivity.kt
package com.example.text

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val button = findViewById<Button>(R.id.button)

        val textComponent = findViewById<TextView>(R.id.textlabel)
        var isClicked = false

        button.setOnClickListener {
            if (isClicked == false) {
                textComponent.text = "HI"
                isClicked = true
            }
            else {
                textComponent.text = "HELLO"
                isClicked = false
            }
        }
    }
}
